java/runtime/src/main/java/org/ray/runtime/AbstractRayRuntime.java

```diff
@@ -88,6 +88,8 @@ public AbstractRayRuntime(RayConfig rayConfig) {
   @Override
   public <T> List<T> get(List<UniqueId> objectIds) {
     boolean wasBlocked = false;
+    // TODO(swang): If we are not on the main thread, then we should generate a
+    // random task ID to pass to the backend.
```

Comment:
I think Java should always have a main thread. that shouldn't matter.

But I think it'd be better to minimize changes and only modify the `RayletClientImpl`, because the fake task id is a temp workaround and shouldn't be part of the `RayletClient` api. 

And you can use a ThreadLocal to hold the fake task id for each thread.

Reply:
Thanks, @raulchen. How would I only modify `RayletClientImpl`? Does it have access to the `WorkerContext`? I'm not familiar enough with the Java code to figure out how all the pieces fit together.

Reply:
@raulchen  We have already implemented the sync issues in multi-threads in our internal version for Java.
maybe we could move it to this.

Reply:
Thanks, @jovan-wang! In the interest of getting this one merged as soon as possible, do you have any suggestions on how to improve the Java changes for this PR?

Reply:
@stephanie-wang 
What we did internally is passing an NIL task id for the main thread (because in node manager you can get the real task id from `worker.GetAssignedTaskId()`), and random task ids for other threads. In this way, you don't need access to `WorkerContext` in `RayletClientImpl`. 

If you still prefer passing the real task id for main thread from worker, you can pass the `runtime` as a parameter to the constructor of `RayletClientImpl` (in `RayNativeRuntime.start`), then you can access `workerContext` by `runtime.getWorkerContext()`.